### PR TITLE
Some small enhancements on gitlog2changelog.py 

### PR DESCRIPTION
**Pull request by xdelaruelle**
I am using the `gitlog2changelog.py` utility from this repository for [my project](https://github.com/cea-hpc/modules). As it is really useful, I am pleased to contribute back by proposing a few enhancements on `gitlog2changelog.py` tools:

* First is to make it resilient to specific git log format configured in user's git configuration.
* Second is to apply code change suggested by Black Python code formatter
* Third is to fix code issues spotted by Flake8 Python utility

### CODE REVIEW

File: tools/gitlog2changelog.py.in [hunk 1]
```diff
@@ -69,35 +86,35 @@ for line in fin:
...
-    elif '    git-svn-id:' in line:
+    elif "    git-svn-id:" in line:
         continue
     # The sign off line is ignored too
-    elif 'Signed-off-by' in line:
+    elif "Signed-off-by" in line:
         continue
     # Extract the actual commit message for this commit
-    elif authorFound & dateFound & messageFound == False:
+    elif authorFound & dateFound & messageFound is False:
```
Comment (jimklimov) [inline tools/gitlog2changelog.py.in:117]: Wondering if `&` here should better become `and`?..

File: tools/gitlog2changelog.py.in [hunk 2]
```diff
@@ -3,16 +3,17 @@
 # Minor changes for NUT by Charles Lepple
 # Distributed under the terms of the GNU General Public License v2 or later
 
-import string, re, os
```
Comment (jimklimov) [inline tools/gitlog2changelog.py.in:6]: Was `string` not needed?

File: tools/gitlog2changelog.py.in (hunk 1 above)
Comment (xdelaruelle) [reply tools/gitlog2changelog.py.in:117]: It was not suggested by either Black and Flake8

**Comment by jimklimov** [comment]
Thanks! I'll take a look at why CI complains when I get to a computer.

Did you have a moment to check if it still works for Python2 (2.7 or even 2.6) and not only 3.x? I have a bit of concern if it already had `r"regex"` notation...

File: tools/gitlog2changelog.py.in (hunk 2 above)
Comment (xdelaruelle) [reply tools/gitlog2changelog.py.in:6]: It seems so. No functions from `string` are used.

Moreover I have tested generated the ChangeLog for Nut and for Modules (my project) before and after these changes: I obtain the exact same file.

**Comment by xdelaruelle** [comment]
> Did you have a moment to check if it still works for Python2 (2.7 or even 2.6) and not only 3.x? I have a bit of concern if it already had `r"regex"` notation...

I have just tested against Python 2.7 and Python 2.6: exact same file is obtained before and after applying these changes to `tools/gitlog2changelog.py`.

Script execution with Python 2.7 is silent (no error message).

With Python 2.6, following informational message is obtained (either with or without the proposed patches applied):

```console
$ /usr/local/python2.6/bin/python2.6 tools/gitlog2changelog.py
Exception AttributeError: "'Popen' object has no attribute '_child_created'" in <bound method Popen.__del__ of <subprocess.Popen object at 0x7ff9cd7c7ed0>> ignored
```

File: tools/gitlog2changelog.py.in (hunk 1 above)
Comment (xdelaruelle) [reply tools/gitlog2changelog.py.in:117]: Sure, I can add another commit to this that point.

Comment (jimklimov) [reply tools/gitlog2changelog.py.in]: Ok, seems to generate same Changelog contents when using the same python interpreter with and without this change :)

**Comment by jimklimov** [comment]
> I have just tested against Python 2.7 and Python 2.6: exact same ChangeLog file is obtained before and after applying these changes to tools/gitlog2changelog.py.

Just to clarify: in your case, did you generate the Changelog with original and updated script using same Python interpreter version and get "exact same" contents in both such cases, or were the contents also identical for different Python versions?

Asking, because I've now tried it with Ubuntu Focal's choice of interpreters, and the resulting files differ for Py2 and Py3 runs with:
````
$ python2 --version
Python 2.7.18

$ python3 --version
Python 3.8.10

$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
````

Differences seem to be due to different end-of-line breaks at dash and slash symbols in filenames... now seeing this, I'd want filenames not broken and lines to only roll over at whitespaces, but that is a separate issue probably, for example:
````
          man/upsrw.8, man/upssched.8, man/upssched.conf.5, man/upsset.cgi.8,
          man/upsset.conf.5, man/upsstats.cgi.8, man/upsstats.html.5,
          man/victronups.8, packaging/RedHat/nut.spec,
-         packaging/RedHat/nut.spec.in, scripts/HP-UX/Makefile, scripts/HP-UX
-         /nut-drvctl, scripts/HP-UX/nut-drvctl.sh, scripts/HP-UX/nut-upsd,
-         scripts/HP-UX/nut-upsd.sh, scripts/HP-UX/nut-upsmon, scripts/HP-UX
-         /nut-upsmon.sh, scripts/README, scripts/RedHat/README,
+         packaging/RedHat/nut.spec.in, scripts/HP-UX/Makefile, scripts/HP-
+         UX/nut-drvctl, scripts/HP-UX/nut-drvctl.sh, scripts/HP-UX/nut-upsd,
+         scripts/HP-UX/nut-upsd.sh, scripts/HP-UX/nut-upsmon, scripts/HP-
+         UX/nut-upsmon.sh, scripts/README, scripts/RedHat/README,
          scripts/RedHat/halt.patch, scripts/RedHat/ups, scripts/RedHat/upsd,
          scripts/RedHat/upsd.in, scripts/RedHat/upsmon,
          scripts/RedHat/upsmon.in, scripts/Solaris8/S99upsmon,
````

**Comment by xdelaruelle** [comment]
I have built the ChangeLog file, from master branch, against the 3 python interpreters (2.6, 2.7 and 3.11). Then moved to this pull request branch, regenerate `tools/gitlog2changelog.py` and build again the ChangeLog against the 3 pythons.

When comparing the before and after file of each interpreter, there is no difference.

When comparing the python2 files, no difference

But there is a difference between ChangeLog generated with python2 (either 2.6 or 2.7) and those generated with python3.

I observe the same difference than you describe and I obtain the exact same difference whether if I use the current tools/gitlog2changelog.py script in master branch or the updated version proposed in this pull request.

This observed difference is I thing due to some fixes on the TextWrapper lib, either:
* https://github.com/python/cpython/commit/72bd327db0b26e542a327449bef77bd2bc059da4
* https://github.com/python/cpython/commit/f3ebc9fe3fae3d44da4d0da9764ed7c033115f12

**Comment by jimklimov** [comment]
Thanks for the finding! Wondering now, if the `TextWrapper` class usage in our scriptware can then be coerced into not breaking the filenames - but that would be a separate PR to tackle :)

**Pull request merged**